lib/Sema/TypeCheckError.cpp

```diff
@@ -679,6 +679,18 @@ class ApplyClassifier {
   Classification classifyRethrowsArgument(Expr *arg, Type paramType) {
     arg = arg->getValueProvidingExpr();
 
+    // If this argument is `nil` literal, it doesn't cause the call to throw.
+    if (arg->isImplicit()) {
```

Comment:
I don't think this needs to be restricted to implicit code; if the user passes `.none`, we should handle that as well.

Reply:
`.none` actually doesn't have this problem, because of the way it's getting matched.

Reply:
But I can remove implicit here.